### DISCUSSION TITLE
How to generate Typescript types for prisma schema

### DISCUSSION THREAD

**Discussion by MatteoGauthier** [prisma/prisma#8237] [Q&A] [answered]
Hey i wanted to know if it is possible to generate typescript types from the prisma schema. I always makes my own type def like this one 
```ts
export type FarmProps = {
  id: number
  name: string
  owner: {
    name: string
    email: string
  } | null
  description: string | null
}
```

But if there was a faster way I'm interested 😛

**Comment by ryands17** [comment, marked as answer]
@MatteoGauthier 👋 

You can use `prisma generate` to generate TypeScript types and import them from `@prisma/client`. You can read more about that [here](https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety).

**Reply by MatteoGauthier** [reply]
Okay Thanks good news ! 🎉

**Reply by MatteoGauthier** [reply]
But i have an issue when trying to access inside an object relation :
![image](https://user-images.githubusercontent.com/32040951/125442838-956e951d-ee10-4d83-9bd2-e66d80e84434.png)
![image](https://user-images.githubusercontent.com/32040951/125445264-940e2d9d-618a-4141-93ca-f3b192a59398.png)

I've imported types for my props `const Farm: React.FC<Prisma.FarmSelect> = (props) => {`

And my schema look like this 

```prisma
model Farm {
  id          Int       @id @default(autoincrement())
  name        String
  description String?   @default("Farm description")
  owner       User?     @relation(fields: [ownerId], references: [id])
  ownerId     Int?
  products    Product[]
}

model User {
  id            Int       @id @default(autoincrement())
  name          String?
  email         String?   @unique
  emailVerified DateTime? @map(name: "email_verified")
  image         String?
  createdAt     DateTime  @default(now()) @map(name: "created_at")
  updatedAt     DateTime  @updatedAt @map(name: "updated_at")
  posts         Post[]
  farms         Farm[]

  CartElement CartElement[]
  @@map(name: "users")
}
```

I still got a typecript error in vscode... (but the data is diplayed on the brower)

If anyone has a solution 🙃

**Reply by ryands17** [reply]
It should be:

```ts
import { Prisma } from '@prisma/client'

type FarmWithOwner = Prisma.FarmGetPayload<{
  include: { owner: true }
}>

const Farm: React.FC<FarmWithOwner> = (props) => {
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
